### ISSUE TITLE
`cc' failed in phase `C Compiler'.

### ISSUE THREAD

**Issue by chris-martin** [skogsbaer/HTF#97]
I'm trying to build HTF 0.14.0.3 with GHC 8.8.1. The error output I'm getting is:

```
[ 5 of 24] Compiling Test.Framework.Preprocessor ( Test/Framework/Preprocessor.hs, dist/build/Test/Framework/Preprocessor.p_o )
cc1: error: unrecognized command line option ‘--layout’
cc1: error: unrecognized command line option ‘--hashes’
cc1: error: unrecognized command line option ‘--cpp’

<no location info>: error:
    `cc' failed in phase `C Compiler'. (Exit code: 1)
```

I'm using the Nix expression that cabal2nix generates for this package, which is:

```
{ mkDerivation, aeson, aeson-pretty, array, base, base64-bytestring
, bytestring, Cabal, containers, cpphs, Diff, directory, filepath
, haskell-src, HUnit, lifted-base, monad-control, mtl, old-time
, pretty, process, QuickCheck, random, regex-compat, stdenv
, template-haskell, temporary, text, time, unix
, unordered-containers, vector, xmlgen
}:
mkDerivation {
  pname = "HTF";
  version = "0.14.0.3";
  sha256 = "17bb9811b15e3bfaf25cda58820509ce1d4cc556eae274e16745742b54810f8d";
  isLibrary = true;
  isExecutable = true;
  setupHaskellDepends = [ base Cabal process ];
  libraryHaskellDepends = [
    aeson array base base64-bytestring bytestring containers cpphs Diff
    directory haskell-src HUnit lifted-base monad-control mtl old-time
    pretty process QuickCheck random regex-compat text time unix vector
    xmlgen
  ];
  libraryToolDepends = [ cpphs ];
  executableHaskellDepends = [
    array base cpphs directory HUnit mtl old-time random text
  ];
  executableToolDepends = [ cpphs ];
  testHaskellDepends = [
    aeson aeson-pretty base bytestring directory filepath HUnit mtl
    process random regex-compat template-haskell temporary text
    unordered-containers
  ];
  homepage = "https://github.com/skogsbaer/HTF/";
  description = "The Haskell Test Framework";
  license = stdenv.lib.licenses.lgpl21;
}
```

Unfortunately I do not really understand the error message, so I apologize if this is not the right place to report this issue. Is there perhaps another non-Haskell dependency that I am missing?

**Comment by skogsbaer** [comment]
It seems like the wrong C preprocessor is called.

**Comment by skogsbaer** [comment]
maybe this is also fixed by https://github.com/skogsbaer/HTF/pull/98/commits/aa541d729e86a17f8f4be530d45feaf20661bdee

**Comment by skogsbaer** [comment]
should be fixed in 0.14.0.4
